### ISSUE TITLE
Get-PackageUpdateInfo ignores ExcludeModuleFromChecking

### ISSUE THREAD

**Issue by tbone4711** [AndiBellstedt/PackageUpdateInfo#33]
v 1.2.3.0
`$modules=Get-InstalledModule`
`$graph=($modules|where {$_.name -like "Microsoft.Graph.*"}).name`
`$az=($modules|where {$_.name -like "Az.*"}).name`
`Set-PackageUpdateSetting -ExcludeModuleFromChecking ($graph+$az)`
`Get-PackageUpdateInfo -ShowOnlyNeededUpdate `

It seems that Get-PackageUpdateInfo only considers ExcludeModuleFromChecking for filtering the output, but not when checking against the gallery. (makes it incredibly slow for Az and Microsoft.Graph with more than 122 sub-modules)

`WARNING: Local version for module 'Az' is higher than online version: local v9.7.1 <-- v10.4.1 online`
`WARNING: Local version for module 'Az.KeyVault' is higher than online version: local v4.9.3 <-- v4.12.0 online`
`WARNING: Local version for module 'Az.Resources' is higher than online version: local v6.6.1 <-- v6.11.1 online`

Because Az.KeyVault and Az.Resources are on the exclude-list and should not be checked. (Only the main module Az is needed)

Regards,
Thomas

P.S. Cloud you add an/more "real live" examples" in the man pages for Set-PackageUpdateSetting? (e.g. excluding Az.* and Mirosoft.Graph.*)

**Comment by AndiBellstedt** [comment]
Hi Thomas,
very happy to see someone using my module and is willing to file me issues here.

### TLDR:
- Version 1.2.5.0 will be available soon, fulfilling your requirements from this issue.
- The displayed warnings with the version missmatches were already reporting in #24  and fixex in version 1.2.4.0

### In detail:
I've considered your requirements and did some changes in the module gathering and filtering inside 'Get-PackageUpdateInfo'.
Following your code example, the performance on checking should be way more faster now. _**BUT, I wouldn't recommend you doing it this way!**_\
Especially for the Az and Graph modules, they are updates so often and you will not get any notice about these updates anymore. Basically, the whole idea for this project is born on these frequently updated modules.
I am well are of the performance impacts with may modules installed, but this is not solvable with my module.\
(also discussed in a [linked in thread](https://www.linkedin.com/feed/update/urn:li:activity:7112342425787846656/)) 

As listed in the [practical use section on main project page](https://github.com/AndiBellstedt/PackageUpdateInfo#practical-usage), I'll suggest to put these two lines in you PSProfile:
- `Start-Job -ScriptBlock { Get-PackageUpdateInfo -ShowOnlyNeededUpdate -ShowToastNotification | Export-PackageUpdateInfo } | Out-Null`
- `Import-PackageUpdateInfo`
This will start the update checking as a background job and show's you quickly the information from the last job run on every startup of the PowerShell.

I've added also two more examples in the manuals on 'Set-PackageUpdateSetting'. 
Actually, you can use the command this way:
- `Set-PackageUpdateSetting -ExcludeModuleFromChecking ("Microsoft.Graph.*", "AZ.*")`
But, of course, your way is fine, also.

**Comment by tbone4711** [comment]
Hi Andi.
Am I correct, with the command when tyrying to include the main-module Microsft.Graph.Beta?
`Set-PackageUpdateSetting -ExcludeModuleFromChecking ("Microsoft.Graph.*", "AZ.*") -IncludeModuleForChecking "Microsoft.Graph.Beta"`
Should result in only showing updates of the 3 main-modules Az, Microsoft.Graph and Microsoft.Graph.Beta.
Regards,
Thomas
P.S. Thanx for you module, which saves me a lot of time on various systems I've to deal with.

**Comment by AndiBellstedt** [comment]
Hi Thomas,
I think this will not fit your needs.\
In this special usecase (having the stable and beta version both on the machine and only want to check for updates on the beta), you have to use the way you outlined in your [first commend of this issue](https://github.com/AndiBellstedt/PackageUpdateInfo/issues/33#issue-1917872458) but without the `-IncludeModuleForChecking` switch.

The rule system for UpdateChecking contains a '*' as a default rule. That means every module is checked on available updates.\
If you use `Set-PackageUpdateSetting-IncludeModuleForCheckin Microsoft.Graph.Beta` you tell the rulesystem only to evalute this exacly 1 module to be checked.\
Combined with the exclusions with the wildcard, every module that fit's the name `"Microsoft.Graph.*"` will be sorted out from checking afterwards.\
\
So for your example `Set-PackageUpdateSetting -ExcludeModuleFromChecking ("Microsoft.Graph.*", "AZ.*") -IncludeModuleForChecking "Microsoft.Graph.Beta"` will effectivly does not check for any module updates at all.

If you'd like to check only for updates of "Microsoft.Graph.Beta", you can suppress the exclude switch and only specify this exact module:
`Set-PackageUpdateSetting -ExcludeModuleFromChecking "" -IncludeModuleForChecking "Microsoft.Graph.Beta"`

Be adviced, NO other module (PSReadline, PackageManagement, PackageUpdateInfo it self) will be checked for updates with this configuration.

**Comment by AndiBellstedt** [comment]
thinking again on your example, maybe this fit's your need:
```
$modules = Get-InstalledModule
$graph = ($modules | Where-Object Name -like "Microsoft.Graph.*" | Where-Object name -notlike "Microsoft.Graph.Beta").name
Set-PackageUpdateSetting -ExcludeModuleFromChecking $graph
Get-PackageUpdateInfo -ShowOnlyNeededUpdate
```
